### ISSUE TITLE
[BUG] When run script loglevel is ignored unless set to "silent"

### ISSUE THREAD

**Issue by crystalfp** [npm/cli#3354]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Running any script by npm shows a log of the script execution even if it is not an error. For example:
```
npm run my-script --loglevel error

> my_server@0.19.2 my-script D:\Project
> npx copyfiles --flat client/*.png server/http
```
The only way to silence the log message is by setting `--loglevel silent`

The problem is that setting loglevel in .npmrc silences also valid messages from npm. I had to spent too much time to debug why npm was not installing my application. Removing .npmrc revealed a version problem in my packages.

BTW this issue is a re-submission for npm@7 of #2468 for npm@6

### Expected Behavior

If it is not an error, nothing should be shown by npm, only the script messages should appear.

### Steps To Reproduce

- Create a directory with a project in it
- Edit package.json and add script `"start": "node -e console.log('Hello')"`
- Run `npm start`. It shows the log
- Run `npm start --loglevel error`. It show the log message, but it should not, it is not an error
- Run `npm start --loglevel silent`. No log message.

### Environment

- OS: Windows 10 64bits
- Node: 16.2.0
- npm: 7.15.1

**Comment by ruyadorno** [comment]
hi @crystalfp thanks for taking the time to reporting this, there are a few other loglevel/output issues that this makes me think of that we were currently tackling:

- https://github.com/npm/cli/pull/4335
- npm/cli#4724

From my point of view it seems to me that tweaking `npm run` to only print errors (maybe only print the stderr stream from the child process?) is a reasonable thing to do.

@lukekarrys I'd love to hear your thoughts on it, I know you've been handling the overall output clean up so this one is a very good one to keep in mind.

**Comment by crystalfp** [comment]
Well, I got into the habit of using always the `-s` option. So in case of strange behaviors, I simply rerun npm without it. Having a functioning loglevel option (or something similar) that I can put in the .npmrc file and forget it is sure much simpler.

**Comment by lukekarrys** [comment]
In bcc781abf10b46aa5b04da996fc4602524be9b0e, this output was moved from stdout to stderr when in `--json` mode.

I think this should probably be what happens all the time, but that would be a breaking change. I documented the new behavior with a comment in the code indicating this should be changed in the future. I think these should probably be log messages with a level of `info` (but maybe still rendered with a `>` prefix) so they can be hidden with the standard loglevel filters.

https://github.com/npm/cli/blob/b54cdb836d6c4146a1aa8e1a5fe9655ba2ed0a6a/lib/utils/display.js#L321-L326
